File: 0455_assign_cookies.py
# https://leetcode-cn.com/problems/assign-cookies/

class Solution:
    def findContentChildren(self, g, s):
        """
        Inputs:
            g: list[int]
            s: list[int]
        Outputs:
            res: int
        """

        g.sort()
        s.sort()

        res = 0
        idx = 0
        n = len(s)
        for child in g:
            while idx < n and child > s[idx]:
                idx = idx + 1
            if idx >= n:
                break

            res = res + 1
            idx = idx + 1
        return res

File: Cpp/0455_assign_cookies.cpp
// https://leetcode.cn/problems/assign-cookies/

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end(),
            [](const int g1, const int g2) {
                return g1 < g2;
            });
        sort(s.begin(), s.end(),
            [](const int s1, const int s2) {
                return s1 < s2;
            });
        int res = 0;
        int i = 0, n = s.size();
        for (int child: g) {
            while ((i < n) && (child > s[i])) {
                i += 1;
            }
            if (i == n) {
                break;
            }
            else {
                res += 1;
                i += 1;
            }
        }
        return res;
    }
};
